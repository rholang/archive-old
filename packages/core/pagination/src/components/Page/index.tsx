import React, { Component } from 'react';
import Button, { ButtonProps, ButtonAppearances } from '@atlaskit/button';

type Diff<T, U> = T extends U ? never : T;

type PagePropsType = Diff<
  ButtonProps,
  {
    appearance?: ButtonAppearances;
    autoFocus: boolean;
    isDisabled: boolean;
    isLoading: boolean;
    spacing: 'compact' | 'default' | 'none';
    shouldFitContainer: boolean;
    type: 'button' | 'submit';
  }
> & { page?: any };

export default class Page extends Component<PagePropsType> {
  render() {
    return <Button {...this.props} appearance="subtle" />;
  }
}
