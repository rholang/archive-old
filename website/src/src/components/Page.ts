import styled from 'styled-components';
import { gridSize, math } from '@atlaskit/theme';

const PageContainer = styled.main`
  width: 100vw;
  max-width: calc(640px + 4rem);
  box-sizing: border-box;
  margin: 2rem auto;
  padding: 0 2rem;
`;

export default PageContainer;

export const Title = styled.h1`
  margin-bottom: 1em;
`;

export const Section = styled.section`
  margin-top: 3em;

  p {
    line-height: 1.4em;
  }
`;

export const Intro = styled.p`
  font-size: ${math.multiply(gridSize, 2)}px;
  font-weight: 300;
  line-height: 1.4em;
`;
