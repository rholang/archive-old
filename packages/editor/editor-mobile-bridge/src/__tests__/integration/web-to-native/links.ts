import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';
import constant from 'lodash.constant';
import times from 'lodash.times';

import {
  callNativeBridge,
  editor,
  editable,
  skipBrowsers as skip,
  getBridgeOutput,
} from '../_utils';

BrowserTestCase(
  'currentSelection when no selection',
  // Safari has issues with key events
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);
    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await browser.type(editable, 'Normal Text');
    await browser.keys('ArrowLeft');

    const currentSelection = await getBridgeOutput(
      browser,
      'linkBridge',
      'currentSelection',
    );
    expect(currentSelection).toMatchCustomSnapshot(testName);
  },
);

BrowserTestCase(
  'currentSelection when selection',
  // Safari has issues with key events
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);
    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await browser.type(editable, 'Normal Text');
    await browser.keys([
      ...times(4, constant('ArrowLeft')),
      'Shift',
      ...times(4, constant('ArrowRight')),
      'Shift',
    ]);

    const currentSelection = await getBridgeOutput(
      browser,
      'linkBridge',
      'currentSelection',
    );
    expect(currentSelection).toMatchCustomSnapshot(testName);
  },
);

BrowserTestCase(
  'currentSelection when cursor is on link',
  // Safari has issues with key events
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);
    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await browser.type(editable, '');
    await callNativeBridge(
      browser,
      'onLinkUpdate',
      'Atlassian',
      'https://www.google.com',
    );
    await browser.keys('ArrowLeft');

    const currentSelection = await getBridgeOutput(
      browser,
      'linkBridge',
      'currentSelection',
    );
    expect(currentSelection).toMatchCustomSnapshot(testName);
  },
);

BrowserTestCase(
  'currentSelection when cursor is selecting a link',
  // Safari has issues with key events
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);
    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await browser.type(editable, '');
    await callNativeBridge(
      browser,
      'onLinkUpdate',
      'Atlassian',
      'https://www.google.com',
    );

    await browser.keys([
      ...times(9, constant('ArrowLeft')),
      'Shift',
      ...times(9, constant('ArrowRight')),
      'Shift',
    ]);

    const currentSelection = await getBridgeOutput(
      browser,
      'linkBridge',
      'currentSelection',
    );
    expect(currentSelection).toMatchCustomSnapshot(testName);
  },
);

BrowserTestCase(
  'currentSelection when cursor is selecting text and link',
  // Safari has issues with key events
  { skip: skip.concat('safari') },
  async (client: any, testName: string) => {
    const browser = new Page(client);
    await browser.goto(editor.path);
    await browser.waitForSelector(editable);

    await browser.type(editable, 'This is a ');
    await callNativeBridge(
      browser,
      'onLinkUpdate',
      'link',
      'https://www.google.com',
    );

    await browser.keys([
      ...times(10, constant('ArrowLeft')),
      'Shift',
      ...times(10, constant('ArrowRight')),
      'Shift',
    ]);

    const currentSelection = await getBridgeOutput(
      browser,
      'linkBridge',
      'currentSelection',
    );

    expect(currentSelection).toMatchCustomSnapshot(testName);
  },
);
