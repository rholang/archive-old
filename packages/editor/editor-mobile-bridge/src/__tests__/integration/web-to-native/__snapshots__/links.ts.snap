// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currentSelection when cursor is on link 1`] = `
Array [
  Object {
    "bottom": 53,
    "left": 16,
    "right": 16,
    "text": "Atlassian",
    "top": 34,
    "url": "https://www.google.com",
  },
]
`;

exports[`currentSelection when cursor is selecting a link 1`] = `
Array [
  Object {
    "bottom": 53,
    "left": 16,
    "right": 16,
    "text": "Atlassian",
    "top": 34,
    "url": "https://www.google.com",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": 53,
    "left": 16,
    "right": 16,
    "text": "Atlassian",
    "top": 34,
    "url": "https://www.google.com",
  },
]
`;

exports[`currentSelection when cursor is selecting text and link 1`] = `
Array [
  Object {
    "bottom": 53,
    "left": 67.84375,
    "right": 67.84375,
    "text": "link",
    "top": 34,
    "url": "https://www.google.com",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": " ",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": " i",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": " is",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": " is ",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": " is a",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": " is a ",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "",
    "top": -1,
    "url": "",
  },
]
`;

exports[`currentSelection when no selection 1`] = `Array []`;

exports[`currentSelection when selection 1`] = `
Array [
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "T",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "Te",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "Tex",
    "top": -1,
    "url": "",
  },
  Object {
    "bottom": -1,
    "left": -1,
    "right": -1,
    "text": "Text",
    "top": -1,
    "url": "",
  },
]
`;
